nevergrad/optimization/base.py

```diff
@@ -133,6 +133,13 @@ def __init__(
         # to make optimize function stoppable halway through
         self._running_jobs: tp.List[tp.Tuple[p.Parameter, tp.JobLike[tp.Loss]]] = []
         self._finished_jobs: tp.Deque[tp.Tuple[p.Parameter, tp.JobLike[tp.Loss]]] = deque()
+        self._max_num_warnings = 50  # No use printing more than 50 warnings.
+
+    def warning(self, s: str, e: tp.Any) -> None:
```

Comment:
this should be a private method (`_warn`)
and instead of a number, I guess I would rather there was one warning per warning type, this may provide more diverse information